### DISCUSSION TITLE
How to configure pub sub?

### DISCUSSION THREAD

**Discussion by erickne** [imdhemy/laravel-in-app-purchases#127] [Q&A] [answered]
**Describe the bug**
Can't find pub/sub information to validade purchases.

**To Reproduce**
1. Purchase a newsubscription or product using google IAP in react native
2. Purchase processe is successfull in mobile app
3. Nothing happens in backend

When using Apple IAP, everything is fine and configured.

**Expected behavior**
There should have some information in pub/sub?

**Full trace**
After purchasing and product or subscription, nothing happens in my pub/sub sextion in Google Console.

**Additional context**

**Comment by vesper8** [comment, marked as answer]
This was a bit tricky to set up. I used the instructions here which are very complete and detailed. Even if you don't use that service, the instructions are still the ones you want to follow:

https://billing.fovea.cc/documentation/connect-with-google-publisher-api/#2_Enable_Real-time_developer_notifications_for_your_app

A very important step which seems to have a lot of people confused is the 4th step inside 2.2:

```

Then fill in the form with those values:
Member: google-play-developer-notifications@system.gserviceaccount.com
Role: Pub/Sub Publisher
```

It's important that you specifically assign the permission to `google-play-developer-notifications@system.gserviceaccount.com`

This is a special system email and this IS the one that you must use. I was struggling with this and kept using my own service-account email, but this is wrong and the above email is the one you must use, everyone needs to use that one. After I did this I was able to send a test notification and was finally able to start receiving them on my laravel backend using this library.
